kick/categories.py

```diff
@@ -1,16 +1,176 @@
+    """
+    A dataclass representing search highlight information
+
+    Attributes
+    -----------
+    field: str
+        The field that was highlighted
+    matched_tokens: list[str]
+        List of tokens that matched
+    snippet: str
+        The highlighted text snippet
+    """
+
+    @property
+    def field(self) -> str:
+        return self._data["field"]
+
+    @property
+    def matched_tokens(self) -> list[str]:
+        return self._data["matched_tokens"]
+
+    @property
+    def snippet(self) -> str:
+        return self._data["snippet"]
+
+    def __repr__(self) -> str:
+        return f"<SearchHighlight field={self.field!r} matched_tokens={self.matched_tokens!r} snippet={self.snippet!r}>"
+
+
+class TextMatchInfo(BaseDataclass["TextMatchInfoPayload"]):
+    """
+    A dataclass representing text match scoring information
+
+    Attributes
+    -----------
+    best_field_score: str
+        Score of the best matching field
+    best_field_weight: int
+        Weight of the best matching field
+    fields_matched: int
+        Number of fields that matched
+    num_tokens_dropped: int
+        Number of tokens that were dropped
+    score: str
+        Overall match score
+    tokens_matched: int
+        Number of tokens that matched
+    typo_prefix_score: int
+        Score for typo/prefix matches
+    """
+
+    @property
+    def best_field_score(self) -> str:
+        return self._data["best_field_score"]
+
+    @property
+    def best_field_weight(self) -> int:
+        return self._data["best_field_weight"]
+
+    @property
+    def fields_matched(self) -> int:
+        return self._data["fields_matched"]
+
+    @property
+    def num_tokens_dropped(self) -> int:
+        return self._data["num_tokens_dropped"]
+
+    @property
+    def score(self) -> str:
+        return self._data["score"]
+
+    @property
+    def tokens_matched(self) -> int:
+        return self._data["tokens_matched"]
+
+    @property
+    def typo_prefix_score(self) -> int:
+        return self._data["typo_prefix_score"]
+
+    def __repr__(self) -> str:
+        return f"<TextMatchInfo score={self.score!r} tokens_matched={self.tokens_matched} fields_matched={self.fields_matched}>"
+
+
+class CategorySearchHit(BaseDataclass["CategorySearchHitPayload"]):
+    """
+    A dataclass representing a category search hit result
+
+    Attributes
+    -----------
+    document: SearchCategory
+        The matching category document
+    highlight: dict[str, TextHighlight]
+        Highlights for each field
+    highlights: list[SearchHighlight]
+        List of all highlights
+    text_match: int
+        Text match score
+    text_match_info: TextMatchInfo
+        Detailed text match information
+    """
+
+    @cached_property
+    def document(self) -> SearchCategory:
+        return SearchCategory(data=self._data["document"])
+
+    @cached_property
+    def highlight(self) -> dict[str, TextHighlight]:
```

Comment:
I don't really like `highlight` and `highlights` because they give me impressions that are different than their description. Not sure what would be better though